### PR TITLE
Bugfix/polyfill outdated and too much

### LINKED ISSUE #370: Which version of custom elements do we polyfill?
~~I just read at https://www.webcomponents.org/polyfills#custom-elements-polyfill that the polyfill for **Custom Elements V1** is still in progress:~~
> The Custom Elements polyfill provides support for v0 of the Custom Elements spec. A polyfill for v1 is in progress at https://github.com/webcomponents/custom-elements.

~~So I'm a bit confused now, with which API's we have to deal with 🤔 **V0** oder **V1**?
Above mention **V1** polyfill has a [huge list of known limitations and bugs](https://github.com/webcomponents/custom-elements#known-bugs-and-limitations)~~

**Edit:**
Okay seems the official site is out of date.
The github repo provides **V1** polyfills as can be read here:
https://github.com/WebComponents/webcomponentsjs#webcomponentsjs-v1-spec-polyfills

And here is an additional list of known issues:
https://github.com/WebComponents/webcomponentsjs#known-issues

### LINKED ISSUE #371: We only need to load the custom elements polyfill
## Are you reporting a bug or a feature?

 - [x] Bug

## Expected Behavior

Since we decided to only use **Custom Elements V1** we just need to load this specific polyfill.
https://github.com/webcomponents/custom-elements

**Note:**
We need to make sure that `WebComponentsReady` event is still fired.

## Actual Behavior

We currently load [`webcomponents-lite.js` ](https://github.com/WebComponents/webcomponentsjs#how-to-use) which includes all polyfills:
> all of the polyfills: HTML Imports, Custom Elements, Shady DOM/CSS and generic platform polyfills (such as ES6 Promise, Constructable events, etc.) (needed by Internet Explorer 11), and Template (needed by IE 11 and Edge)

screen shot 2018-03-27 at 16 15 02

### LINKED ISSUE #505: vertx dependency in webcomponents-lite.js broken for consumers
## Are you reporting a bug or a feature?

 - [x] Bug
 - [ ] Feature

## Expected Behavior

Webcomponents polyfill should work for consumers.

So I suggest that we remove this from our build and let our users load it directly.

## Actual Behavior

`Module not found: Can't resolve 'vertx'`

https://github.com/webcomponents/webcomponentsjs/issues/794

### PR DESCRIPTION
**Pull request by AndyOGo**
Fixes #505 
Fixes #371 
Fixes #370 
Improves #225 #172 

Changes proposed in this pull request:

 - only `custom-elements` polyfill
 - fixing `withReact` to wait for [`customElements.whenDefined`](https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry/whenDefined)
 - removed copy-polyfills script for webcomponents
- updated webcomponents polyfill
- updated readme

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
 
 # How Has This Been Tested?
 
 
 
 # Checklist:
 
 - [x] My code follows the style guidelines of this project
 - [x] I have performed a self-review of my own code
 - [x] I have commented my code, particularly in hard-to-understand areas
 - [x] I have made corresponding changes to the documentation
 - [x] My changes generate no new warnings
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by AndyOGo** [comment]
did you run `npm install`?
I updated `@webcomponents` from v1 to v2 and there are breaking changes of file paths

**Comment by AndyOGo** [comment]
```sh
[1] [3] > @axa-ch/patterns-library@2.0.1-beta.160 watch-svg /Users/axawinterthur/dev/axa-ch_style-html-guide
[1] [3] > npm-watch build-icons-dev
[1] [3]
[1] [1] module.js:557
[1] [1]     throw err;
[1] [1]     ^
[1] [1]
[1] [1] Error: Cannot find module '/Users/axawinterthur/dev/axa-ch_style-html-guide/node'
[1] [1]     at Function.Module._resolveFilename (module.js:555:15)
[1] [1]     at Function.Module._load (module.js:482:25)
[1] [1]     at Function.Module.runMain (module.js:701:10)
[1] [1]     at startup (bootstrap_node.js:193:16)
[1] [1]     at bootstrap_node.js:617:3
[1] [1] npm ERR! code ELIFECYCLE
npm ERR! errno 1
[1] [1] npm ERR! @axa-ch/patterns-library@2.0.1-beta.160 watch-js: `node node stack/tasks/watch-js.js`
[1] [1] npm ERR! Exit status 1
[1] [1] npm ERR!
[1] [1] npm ERR! Failed at the @axa-ch/patterns-library@2.0.1-beta.160 watch-js script.
[1] [1] npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
[1] [1]
[1] [1] npm ERR! A complete log of this run can be found in:
[1] [1] npm ERR!     /Users/axawinterthur/.npm/_logs/2018-07-05T09_59_37_923Z-debug.log
[1] [1] npm run watch-js exited with code 1
```

**Comment by AndyOGo** [comment]
IE11 seems to work:

screen shot 2018-07-05 at 12 20 09

**Comment by AndyOGo** [comment]
EDGE 17 is working (though the icons aren't displayed) 🤔 I also have a very bad inet connection...

screen shot 2018-07-05 at 12 29 07

**Comment by AndyOGo** [comment]
hmh, we some Syntax Error in Edge 17 for `AXAHeaderMobileLanguages`:

screen shot 2018-07-05 at 12 31 09

screen shot 2018-07-05 at 12 33 00

**Comment by AndyOGo** [comment]
@LucaMele Got it. Icons are there :)

screen shot 2018-07-05 at 12 40 04

**Comment by AndyOGo** [comment]
And finally even IE11 😲 🚀 

screen shot 2018-07-05 at 12 43 04

**Comment by AndyOGo** [comment]
Even the whole molecules pages displays now at ie11, yeaahhh 😄 

screen shot 2018-07-05 at 12 47 54

**Comment by LucaMele** [comment]
IE React broken and EDGE same slow on my Browserstack account :(

File: src/js/with-react.js [hunk 1]
```diff
@@ -59,19 +61,36 @@ const withReact = (WebComponent, { pure = true, passive = false } = {}) => {
...
 
       this._eventCache = {};
+
+      this.state = {
+        isDefined: false,
+      };
+
+      // make sure that the custom element is ready
+      // ref: https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry/whenDefined
+      customElements.whenDefined(tagName).then(() => {
```
Comment (AndyOGo) [inline src/js/with-react.js]: IE11 does not have `Promise`

screen shot 2018-07-05 at 13 14 16

**Comment by LucaMele** [comment]
Then maybe the `/Users/lucamele/git/patterns-library/dist/app/es6-polyfills.js` is not loaded anymore

File: src/app/es6-polyfills.js [hunk 2]
```diff
@@ -1,2 +1,3 @@
+import '@webcomponents/webcomponentsjs/bundles/webcomponents-ce';
```
Comment (LucaMele) [inline src/app/es6-polyfills.js]: Keep IE polyfills separate from Custom Element ones

Comment (LucaMele) [reply src/app/es6-polyfills.js]: ergo copy paste the webcomponents-ce SEPARATLY to es6-polyfills.js

File: src/app/es6-polyfills.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
+// Reflect is needed to make Custom Element ES6 Classes working
```
Comment (LucaMele) [inline src/app/es6-polyfills.js]: all browser apart of IE11 i think have it nativly so i think also this is just for IE

Comment (AndyOGo) [reply src/app/es6-polyfills.js]: oh okay, will add it

**Pull request merged**